Sources/AppcuesKit/Presentation/ExperienceRendering/ExperienceStateMachine.swift

```diff
@@ -312,10 +332,26 @@ extension ExperienceStateMachine {
                 SdkMetrics.renderStart(experience.requestID)
 
                 presentStep { error in
-                    // on failure, report a fatal error and dismiss the experience
-                    try? machine.transition(.reportError(.step(experience, stepIndex, "\(error)"), fatal: true))
+                    handlePresentationError(error, machine: machine, experience: experience, stepIndex: stepIndex, package: package)
                 }
             }
         }
+
+        private func handlePresentationError(
+            _ error: Error,
+            machine: ExperienceStateMachine,
+            experience: ExperienceData,
+            stepIndex: Experience.StepIndex,
+            package: ExperiencePackage
+        ) {
+            let recoverable = (error as? AppcuesTraitError)?.recoverable ?? false
+            let retryEffect: ExperienceStateMachine.SideEffect? = recoverable ? .retryPresentation(experience, stepIndex, package) : nil
```

Comment:
sort of subtle here, but a `nil` retryEffect basically means it was a fatal error and things conclude. If non-nil, it will go into `.failure` mode and hope for recovery.